README.md

```diff
@@ -208,7 +208,35 @@ At a minimum, all Roaring implementations should be able to parse the two files
 
 ## Unsigned integers in Java
 
-
 Java lacks native unsigned integers, but integers are still considered to be unsigned within Roaring, and ordered  according to  ``Integer.compareUnsigned``. 
 
 For integers in [0,2147483647], the unsigned and signed integers are undistinguisable since Java relies a 32-bit two's complement format for its ``int`` type.
+
+# Extention for 64B implementations
+
+Some Roaring bitmap implementations may offer a 64-bits implementation. This section proposes a portable format, compatible with some (but not all) 
+64-bits implementations. This format is naturally compatible with implementations based on a conventional red-black-tree (as the serialization format 
+is similar to the in-memory layout). The keys would be 32-bit integers representing the most significant 32~bits of elements whereas the values of the 
+tree are 32-bit Roaring bitmaps. The 32-bit Roaring bitmaps represent the least significant bits of a set of elements.
+
+## General layout
+
+All words are written using little endian encoding.
+
+- Write as long/uint64 the distinct number of buckets (restricted to integers in [0,4294967295]) (a.k.a the number of distinct keys being the most significant 32~bits of elements, leading to a two zero bytes padding).
+- Iterate through buckets ordered by increasing keys (as unsigned integers), for each: 
+- first writing as int/uint32 the most significant 32~bits of the bucket (in [0,2^32-1])
```

Comment:
If we want this memory-mappable, should we store all the keys first together with the offset into the serialised data where their bitmap starts, so that one can basically just read the header and then pick up one particular bitmap by scanning from the offset?

Reply:
(processing comments one at a time).
Yes ! I suggest keeping this feature for a next iteration (as here, I'm mainly targetting portability between C, Go and Java).

The 4 bytes padding may be used for some sort of versioning (or serialization-format picking), as anyway, it may not be reachable to have an optimal format for both Map-based and ART-based 64-bit implementations.

Reply:
Yes, that makes sense. The format we use in Delta has a "magic number" in that slot, which we use to identify the actual format used to serialise the data.